homeassistant/components/yale_smart_alarm/alarm_control_panel.py

```diff
@@ -75,51 +82,91 @@ async def async_setup_entry(
-        return STATE_MAP.get(self.coordinator.data["alarm"]) is not None
-
-    @property
-    def code_arm_required(self):
-        """Whether the code is required for arm actions."""
-        return False
-
-    @property
-    def supported_features(self) -> int:
-        """Return the list of supported features."""
-        return SUPPORT_ALARM_ARM_HOME | SUPPORT_ALARM_ARM_AWAY
-
-    def alarm_disarm(self, code=None):
+    async def async_alarm_disarm(self, code=None) -> None:
         """Send disarm command."""
-        self.coordinator.yale.disarm()
-
-    def alarm_arm_home(self, code=None):
+        if self._coordinator.yale:
+            try:
+                alarm_state = await self.hass.async_add_executor_job(
+                    self._coordinator.yale.disarm
+                )
+            except (
+                AuthenticationError,
+                ConnectionError,
+                TimeoutError,
+                UnknownError,
+            ) as error:
+                raise HomeAssistantError(
+                    f"Could not verify disarmed for {self._attr_name}: {error}"
+                ) from error
+
+        if alarm_state:
+            self._attr_state = STATE_MAP.get(YALE_STATE_DISARM)
+            self.async_write_ha_state()
+        LOGGER.debug("Alarm disarmed: %s", alarm_state)
+
+    async def async_alarm_arm_home(self, code=None) -> None:
         """Send arm home command."""
-        self.coordinator.yale.arm_partial()
-
-    def alarm_arm_away(self, code=None):
+        if self._coordinator.yale:
+            try:
+                alarm_state = await self.hass.async_add_executor_job(
+                    self._coordinator.yale.arm_partial
+                )
+            except (
+                AuthenticationError,
+                ConnectionError,
+                TimeoutError,
+                UnknownError,
+            ) as error:
+                raise HomeAssistantError(
+                    f"Could not verify armed home for {self._attr_name}: {error}"
+                ) from error
+
+        if alarm_state:
+            self._attr_state = STATE_MAP.get(YALE_STATE_ARM_PARTIAL)
+            self.async_write_ha_state()
+        LOGGER.debug("Alarm armed home: %s", alarm_state)
+
+    async def async_alarm_arm_away(self, code=None) -> None:
         """Send arm away command."""
-        self.coordinator.yale.arm_full()
+        if self._coordinator.yale:
+            try:
+                alarm_state = await self.hass.async_add_executor_job(
+                    self._coordinator.yale.arm_full
+                )
+            except (
+                AuthenticationError,
+                ConnectionError,
+                TimeoutError,
+                UnknownError,
+            ) as error:
+                raise HomeAssistantError(
+                    f"Could not verify armed away for {self._attr_name}: {error}"
+                ) from error
+
+        if alarm_state:
```

Comment:
What are the cases where these functions return False?  Are they cases that are worth informing the user about or OK to ignore?

Reply:
Did not consider that but it's not a bad idea maybe to raise an error here if the arming is not succesful so the user gets some more feedback than purely the arming isn't happening.